src/main/java/net/sf/rails/ui/swing/AutoLoadPoller.java

```diff
@@ -13,91 +13,102 @@
 
 
 public class AutoLoadPoller extends Thread {
-    
-    private GameUIManager guiMgr;
+
+    private final GameUIManager guiMgr;
     private String saveDirectory;
     private String savePrefix;
     private String ownPostfix;
     private int pollingInterval;
     private int pollingStatus;
-    
+
     private boolean pollingActive = false;
-    
-    private String lastSavedFilenameFilepath;
+
+    private final String lastSavedFilenameFilepath;
     private String lastSavedFilename = "";
-    
+
     public static final int OFF = 0;
     public static final int ON = 1;
     public static final int SUSPENDED = 2;
 
-    protected static Logger log =
-        LoggerFactory.getLogger(AutoLoadPoller.class);
+    protected static Logger log = LoggerFactory.getLogger(AutoLoadPoller.class);
 
     public AutoLoadPoller (GameUIManager guiMgr, String saveDirectory, String savePrefix, String ownPostfix,
             int status, int pollingInterval) {
- 
         this.guiMgr = guiMgr;
         this.saveDirectory = saveDirectory;
         this.savePrefix = savePrefix;
         this.ownPostfix = ownPostfix;
         this.pollingStatus = status;
         this.pollingInterval = pollingInterval;
-        
+
         lastSavedFilenameFilepath = saveDirectory + "/" + savePrefix + ".last_rails";
-        
-        log.debug("Poller own postfix: "+ownPostfix);
-        log.debug("Poller last-filename path: "+lastSavedFilenameFilepath);
-        
+
+        log.debug("Poller own postfix: {}", ownPostfix);
+        log.debug("Poller last-filename path: {}", lastSavedFilenameFilepath);
     }
 
     @Override
     public void run () {
-
         log.info ("AutoLoadPoller started");
 
+        int currentPollInterval = 1;
         int secs, sleepTime;
-        String currentFilename;
 
         for (;;) {
+            secs = Calendar.getInstance().get(Calendar.SECOND);
+            try {
+                sleepTime = 1000 * (currentPollInterval - secs%currentPollInterval);
```

Comment:
You can move the variable declaration inside this line

Reply:
I didn't touch any of the existing code unless it was related to the actual changes needed-- I've run into lots of possible changes of this nature but haven't touched them unless I would have touched the actual line in question. Note that the SonarQube PR that I'll be pushing in does start to make these types of changes but man, there are a LOT